### PR TITLE
include thoth grafana dashboard to the moc monitoring

### PR DESCRIPTION
**Pull request by harshad16**
include thoth grafana dashboard to the moc monitoring
Signed-off-by: Harshad Reddy Nalla <hnalla@redhat.com>

Depends-On: https://github.com/thoth-station/thoth-application/pull/752
Related-to: https://github.com/operate-first/apps/issues/108

Questions:
How to introduce the datasource along with this? 
Referencing this https://github.com/operate-first/apps/issues/59, should i also include the datasource manifest along with these.

### CODE REVIEW

**Comment by 4n4nd** [comment]
@harshad16 Can you specify the [grafana dashboard folder](https://grafana.com/docs/grafana/latest/dashboards/dashboard_folders/) for all your dashboards? You can do that by adding a field `CustomFolderName` in your dashboard spec ([source](https://github.com/integr8ly/grafana-operator/blob/master/documentation/dashboards.md#dashboard-folder-support)). 
I suggest doing this in your kustomization so it is applied to all you dashboard resources.

File: odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-kafka-argo-metrics.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: integreatly.org/v1alpha1
+kind: GrafanaDashboard
+metadata:
+  name: thoth-kafka-argo-metrics
+  labels:
+    app: grafana
+spec:
+  name: thoth-kafka-argo-metrics
+  url: https://raw.githubusercontent.com/thoth-station/thoth-application/master/grafana-dashboard/overlays/cnv-prod/thoth-kafka-argo-metrics.json
```
Comment (4n4nd) [inline odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-kafka-argo-metrics.yaml:9]: @harshad16 None of the urls seem to be working for me. I get a 404 error when I try to access them

**Comment by harshad16** [comment]
> Can you specify the [grafana dashboard folder](https://grafana.com/docs/grafana/latest/dashboards/dashboard_folders/) for all your dashboards? You can do that by adding a field `CustomFolderName` in your dashboard spec ([source](https://github.com/integr8ly/grafana-operator/blob/master/documentation/dashboards.md#dashboard-folder-support)).
> I suggest doing this in your kustomization so it is applied to all you dashboard resources.

thanks i will include the foldername :+1:

File: odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-kafka-argo-metrics.yaml (hunk 1 above)
Comment (harshad16) [reply odh/overlays/moc/monitoring/dashboards/thoth-station/thoth-kafka-argo-metrics.yaml:9]: yes, as described on the pull request description they depend upon another PR on thoth-application repository.
once that is merged , these URL will be available and i will unhold this pull request .

**Comment by 4n4nd** [comment]
>  How to introduce the datasource along with this?

@harshad16 maybe you can use the [datasources parameter in the dashboardspec](https://github.com/integr8ly/grafana-operator/blob/master/documentation/dashboards.md#datasource-inputs)?

**Review by anishasthana — APPROVED** [review]
Changes look good to me outside of the broken links

**Comment by harshad16** [comment]
/unhold

The dependent pr was merged during the weekend
the links are all available.

**Comment by harshad16** [comment]
>  you can use the [datasources parameter in the dashboardspec](https://github.com/integr8ly/grafana-operator/blob/master/documentation/dashboards.md#datasource-inputs)?

Thanks for pointing this out
should i include the datasource manifest in this pull request or create another one?

**Comment by anishasthana** [comment]
> Thanks for pointing this out
> should i include the datasource manifest in this pull request or create another one?

I think including it in this PR is fine. It will make it more complete

**Comment by harshad16** [comment]
Included the datasource manifests in the folder: datasource 
`odh/overlays/moc/monitoring/datasource/thoth-station-datasource.yaml`
please do let me know if any changes are to be made.

**Comment by 4n4nd** [comment]
@harshad16 looks great! can you just add identifying labels to thoth specific objects?

**Comment by harshad16** [comment]
> can you just add identifying labels to thoth specific objects?

Added the labels `component: thoth-station` to all the manifest

**Comment by harshad16** [comment]
@4n4nd Ready for the review!

**Review by 4n4nd — APPROVED** [review]
LGTM :+1:

**Pull request merged**